dataclients/kubernetes/definitions/routegroups.go

```diff
@@ -392,7 +330,50 @@ func (br *backendReference) validate(backends map[string]bool) error {
 	return nil
 }
 
-func (r *routeSpec) validate(hasDefault bool, backends map[string]bool) error {
+func (r *RouteSpec) UniqueMethods() []string {
+	return uniqueStrings(r.Methods)
+}
+
+func (rg *RouteGroupSpec) validate() error {
+	// has at least one backend:
+	if len(rg.Backends) == 0 {
+		return errRouteGroupWithoutBackend
+	}
+
+	// backends valid and have unique names:
+	backends := make(map[string]bool)
+	for _, b := range rg.Backends {
+		if backends[b.Name] {
+			return backendsWithDuplicateName(b.Name)
+		}
+
+		backends[b.Name] = true
+		if err := b.validate(); err != nil {
+			return invalidBackend(b.Name, err)
+		}
+	}
+
+	hasDefault := len(rg.DefaultBackends) > 0
+	for _, br := range rg.DefaultBackends {
+		if err := br.validate(backends); err != nil {
+			return err
+		}
+	}
+
+	if !hasDefault && len(rg.Routes) == 0 {
+		return errMissingBackendReference
+	}
+
+	for i, r := range rg.Routes {
+		if err := r.validate(hasDefault, backends); err != nil {
+			return invalidRoute(i, err)
+		}
+	}
+
+	return nil
+}
+
+func (r *RouteSpec) validate(hasDefault bool, backends map[string]bool) error {
```

Comment:
can you add a TODO comment that we need to pass `namespace/name` in all errors?

Reply:
Done